src/libsyntax_ext/test.rs

```diff
@@ -234,8 +235,19 @@ fn should_panic(cx: &ExtCtxt<'_>, i: &ast::Item) -> ShouldPanic {
 }
 
 fn has_test_signature(cx: &ExtCtxt<'_>, i: &ast::Item) -> bool {
-    let has_should_panic_attr = attr::contains_name(&i.attrs, sym::should_panic);
     let ref sd = cx.parse_sess.span_diagnostic;
+
+    if attr::contains_name(&i.attrs, sym::test) {
+        cx.parse_sess.buffer_lint(
+            BufferedEarlyLintId::IllFormedAttributeInput,
+            i.span,
+            i.id,
+            "multiple test attributes, only one is needed"
+        );
+        return false;
```

Comment:
We are reporting a warning, not an error, and this is technically legal just suspicious, so we shouldn't return `false` here, but should rather report and continue.
(Also, if `false` is not returned this no longer belongs to `fn has_test_signature` logically.)

Reply:
ok